attestation-agent/lib/src/lib.rs

```diff
@@ -88,6 +88,13 @@ pub trait AttestationAPIs {
         events: Vec<Vec<u8>>,
         register_index: Option<u64>,
     ) -> Result<()>;
+
+    /// Check the initdata binding
+    async fn check_init_data(
+        &mut self,
+        init_data: Vec<u8>,
+        _init_data_plaintext: Option<Vec<u8>>,
```

Comment:
Why is this optional?

Reply:
In the future, we might want to allow carrying the initdata plaintext for more fine-grained check & matches. But current designs we do not need this and only use the initdata digest. I left this API parameter as `Option` for future use.

This once mentioned [the proposal](https://github.com/confidential-containers/confidential-containers/issues/171#issue-1985098907), Step 7 in Technical details

Reply:
Inspired by https://github.com/confidential-containers/guest-components/pull/462/files#r1469814161, I'd make this an enum. Please help to check again if it is good.

Reply:
The link doesn't work for me, but I assume the parameter would be something like this? (pseudocode):

```
enum InitData {
  Hash(Vec<u8>),
  PlainText(Vec<u8>),
  HashAndPlainText({ hash: Vec<u8>, plain_text: Vec<u8>}),
}
```
It might be cumbersome to use, if we really anticipate to pass an init_data body to the function and it doesn't end up being a perpetual dead parameter, I think the Option is fine.

Alternatively, we can also go with the a limited `check_init_data_digest()` fn for now and introduce a more comprehensive `check_init_data()` fn later.